src/UseCases/PushStaleToCyclopsUseCase.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Gcd\Cyclops\UseCases;
+
+use Gcd\Cyclops\Entities\CyclopsCustomerListEntity;
+use Gcd\Cyclops\Exceptions\CyclopsException;
+use Gcd\Cyclops\Services\CyclopsService;
+
+class PushStaleToCyclopsUseCase
+{
+    /**
+     * @var CyclopsService
+     */
+    private $cyclopsService;
+
+    public function __construct(CyclopsService $cyclopsService)
+    {
+        $this->cyclopsService = $cyclopsService;
+    }
+
+    public function execute(CyclopsCustomerListEntity $list)
+    {
+        foreach ($list->items as $item) {
+            try {
+                $this->cyclopsService->setBrandOptInStatus($item, $item->brandOptIn);
+            } catch (CyclopsException $exception) {
```

Comment:
Why are we catching it if we're not going to handle it? Same with **PushDeletedToCyclopsUseCase.php:26**

Reply:
Also if you're getting Forbidden/CustomerNotFound there's no point continuing the **Foreach** because nothing else will be processed anyway.